<a href="https://colab.research.google.com/github/Avazbek-CNU/Breast-cancer-classification/blob/main/obesity_risk.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

In [ ]:
df = pd.read_csv('obesity_level.csv')

In [ ]:
df

id  Gender        Age    Height      Weight  \
0          0    Male  24.443011  1.699998   81.669950   
1          1  Female  18.000000  1.560000   57.000000   
2          2  Female  18.000000  1.711460   50.165754   
3          3  Female  20.952737  1.710730  131.274851   
4          4    Male  31.641081  1.914186   93.798055   
...      ...     ...        ...       ...         ...   
20753  20753    Male  25.137087  1.766626  114.187096   
20754  20754    Male  18.000000  1.710000   50.000000   
20755  20755    Male  20.101026  1.819557  105.580491   
20756  20756    Male  33.852953  1.700000   83.520113   
20757  20757    Male  26.680376  1.816547  118.134898   

       family_history_with_overweight  FAVC      FCVC       NCP        CAEC  \
0                                   1     1  2.000000  2.983297   Sometimes   
1                                   1     1  2.000000  3.000000  Frequently   
2                                   1     1  1.880534  1.411685   Sometimes   
3                                   1     1  3.000000  3.000000   Sometimes   
4                                   1     1  2.679664  1.971472   Sometimes   
...                               ...   ...       ...       ...         ...   
20753                               1     1  2.919584  3.000000   Sometimes   
20754                               0     1  3.000000  4.000000  Frequently   
20755                               1     1  2.407817  3.000000   Sometimes   
20756                               1     1  2.671238  1.971472   Sometimes   
20757                               1     1  3.000000  3.000000   Sometimes   

       SMOKE      CH2O  SCC       FAF       TUE       CALC  \
0          0  2.763573    0  0.000000  0.976473  Sometimes   
1          0  2.000000    0  1.000000  1.000000          0   
2          0  1.910378    0  0.866045  1.673584          0   
3          0  1.674061    0  1.467863  0.780199  Sometimes   
4          0  1.979848    0  1.967973  0.931721  Sometimes   
...      ...       ...  ...       ...       ...        ...   
20753      0  2.151809    0  1.330519  0.196680  Sometimes   
20754      0  1.000000    0  2.000000  1.000000  Sometimes   
20755      0  2.000000    0  1.158040  1.198439          0   
20756      0  2.144838    0  0.000000  0.973834          0   
20757      0  2.003563    0  0.684487  0.713823  Sometimes   

                      MTRANS              0be1dad  
0      Public_Transportation  Overweight_Level_II  
1                 Automobile         0rmal_Weight  
2      Public_Transportation  Insufficient_Weight  
3      Public_Transportation     Obesity_Type_III  
4      Public_Transportation  Overweight_Level_II  
...                      ...                  ...  
20753  Public_Transportation      Obesity_Type_II  
20754  Public_Transportation  Insufficient_Weight  
20755  Public_Transportation      Obesity_Type_II  
20756             Automobile  Overweight_Level_II  
20757  Public_Transportation      Obesity_Type_II  

[20758 rows x 18 columns]

In [ ]:
df.isnull().sum()

id                                0
Gender                            0
Age                               0
Height                            0
Weight                            0
family_history_with_overweight    0
FAVC                              0
FCVC                              0
NCP                               0
CAEC                              0
SMOKE                             0
CH2O                              0
SCC                               0
FAF                               0
TUE                               0
CALC                              0
MTRANS                            0
0be1dad                           0
dtype: int64

In [ ]:
df.columns

Index(['id', 'Gender', 'Age', 'Height', 'Weight',
       'family_history_with_overweight', 'FAVC', 'FCVC', 'NCP', 'CAEC',
       'SMOKE', 'CH2O', 'SCC', 'FAF', 'TUE', 'CALC', 'MTRANS', '0be1dad'],
      dtype='object')

In [ ]:
from sklearn.preprocessing import  LabelEncoder
mk = LabelEncoder()
df['Gender'] = mk.fit_transform(df['Gender'])
df['CALC'] = mk.fit_transform(df['CALC'])
df['MTRANS'] = mk.fit_transform(df['MTRANS'])
df['0be1dad'] = mk.fit_transform(df['0be1dad'])
df['CAEC'] = mk.fit_transform(df['CAEC'])

In [ ]:
df

id  Gender        Age    Height      Weight  \
0          0       1  24.443011  1.699998   81.669950   
1          1       0  18.000000  1.560000   57.000000   
2          2       0  18.000000  1.711460   50.165754   
3          3       0  20.952737  1.710730  131.274851   
4          4       1  31.641081  1.914186   93.798055   
...      ...     ...        ...       ...         ...   
20753  20753       1  25.137087  1.766626  114.187096   
20754  20754       1  18.000000  1.710000   50.000000   
20755  20755       1  20.101026  1.819557  105.580491   
20756  20756       1  33.852953  1.700000   83.520113   
20757  20757       1  26.680376  1.816547  118.134898   

       family_history_with_overweight  FAVC      FCVC       NCP  CAEC  SMOKE  \
0                                   1     1  2.000000  2.983297     3      0   
1                                   1     1  2.000000  3.000000     2      0   
2                                   1     1  1.880534  1.411685     3      0   
3                                   1     1  3.000000  3.000000     3      0   
4                                   1     1  2.679664  1.971472     3      0   
...                               ...   ...       ...       ...   ...    ...   
20753                               1     1  2.919584  3.000000     3      0   
20754                               0     1  3.000000  4.000000     2      0   
20755                               1     1  2.407817  3.000000     3      0   
20756                               1     1  2.671238  1.971472     3      0   
20757                               1     1  3.000000  3.000000     3      0   

           CH2O  SCC       FAF       TUE  CALC  MTRANS  0be1dad  
0      2.763573    0  0.000000  0.976473     2       3        6  
1      2.000000    0  1.000000  1.000000     0       0        0  
2      1.910378    0  0.866045  1.673584     0       3        1  
3      1.674061    0  1.467863  0.780199     2       3        4  
4      1.979848    0  1.967973  0.931721     2       3        6  
...         ...  ...       ...       ...   ...     ...      ...  
20753  2.151809    0  1.330519  0.196680     2       3        3  
20754  1.000000    0  2.000000  1.000000     2       3        1  
20755  2.000000    0  1.158040  1.198439     0       3        3  
20756  2.144838    0  0.000000  0.973834     0       0        6  
20757  2.003563    0  0.684487  0.713823     2       3        3  

[20758 rows x 18 columns]

In [ ]:
df.drop(columns=['id'], inplace=True)

df.iloc[:, :-1] selects all rows (:) and all columns except the last one (:-1) to create the feature variables.

df.iloc[:, -1] selects all rows (:) and only the last column (-1) to create the target variable.
Now, X contains your feature variables, and y contains your target variable.

In [ ]:
X = df.iloc[:, :-1].values
y = df.iloc[:, -1].values

In [ ]:
X

array([[1, 24.443011, 1.699998, ..., 0.976473, 2, 3],
       [0, 18.0, 1.56, ..., 1.0, 0, 0],
       [0, 18.0, 1.71146, ..., 1.673584, 0, 3],
       ...,
       [1, 20.101026, 1.819557, ..., 1.198439, 0, 3],
       [1, 33.852953, 1.7, ..., 0.973834, 0, 0],
       [1, 26.680376, 1.816547, ..., 0.713823, 2, 3]], dtype=object)

In [ ]:
y

array([6, 0, 1, ..., 3, 6, 3])

In [ ]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.25, random_state = 0)

In [ ]:
from sklearn.preprocessing import StandardScaler
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

In [ ]:
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression(random_state = 0)
classifier.fit(X_train, y_train)

/usr/local/lib/python3.10/dist-packages/sklearn/linear_model/_logistic.py:458: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


LogisticRegression(random_state=0)

In [ ]:
y_pred = classifier.predict(X_test)
print(np.concatenate((y_pred.reshape(len(y_pred),1), y_test.reshape(len(y_test),1)),1))

[[4 4]
 [5 5]
 [0 0]
 ...
 [6 2]
 [1 0]
 [4 4]]


In [ ]:
from sklearn.metrics import confusion_matrix, accuracy_score
cm = confusion_matrix(y_test, y_pred)
print(cm)
accuracy_score(y_test, y_pred)

[[ 625   84    3    0    0   58   11]
 [  27  574    0    0    0    2    1]
 [   0    2  608   41    7   17   51]
 [   0    0   33  785    0    0    8]
 [   0    0    0    0 1014    1    0]
 [  64    2   13    0    0  388  116]
 [  14    0   91    7    1   96  446]]


0.8554913294797688

#The model is correct in its predictions approximately 85.5% of the time.